### PR TITLE
Added OCB testing option

### PR DESCRIPTION
**Pull request by aburrell**
This pull requests makes it possible to test the new pysat branch of OCBpy.

- PEP8 changes
- Added OCB tag and sat_id's for the different OCB instruments with appropriate date ranges
- Changed fake file construction to have the same range of dates, but based off of the date provided in test_dates.

### CODE REVIEW

**Comment by jklenzing** [comment]
From Lines 13-14 of `setup` of `TestSpecificUTOrbits` of `test_orbits.py`:
'''
self.testInst = pysat.Instrument('pysat','testing', '86400', clean_level='clean',
                                        orbit_info=info, update_files=True)
'''
The code seems to be trying to run '86400' as a tag, but unable to cope with the new tags in pysat_testing.  Not sure what role '86400' is playing here.

File: pysat/instruments/pysat_testing.py [hunk 1]
```diff
@@ -111,56 +126,66 @@ def load(fnames, tag=None, sat_id=None, sim_multi_file_right=False,
...
 
 def list_files(tag=None, sat_id=None, data_path=None, format_str=None):
     """Produce a fake list of files spanning a year"""
-    
-    index = pds.date_range(pysat.datetime(2008,1,1), pysat.datetime(2010,12,31)) 
-    names = [ data_path+date.strftime('%Y-%m-%d')+'.nofile' for date in index]
+
+    # Determine the appropriate date range for the fake files
+    date_tag = '' if tag not in tags else tag
+    start = test_dates[date_tag][sat_id] - pds.DateOffset(years=1)
```
Comment (rstoneback) [inline pysat/instruments/pysat_testing.py]: Original code covered two years. Also added protection for leap years.

start = test_dates[date_tag][sat_id] - pds.DateOffset(years=1)
stop = test_dates[date_tag][sat_id] + pds.DateOffset(years=2) \
       - pds.DateOffset(days=1)
index = pds.date_range(start, stop)

Comment (rstoneback) [reply pysat/instruments/pysat_testing.py]: I mean three years :)

File: pysat/instruments/pysat_testing.py [hunk 2]
```diff
@@ -85,24 +92,32 @@ def init(self):
     self.new_thing=True
```
Comment (rstoneback) [inline pysat/instruments/pysat_testing.py]: Alternative method for changing file dates that doesn't require overloading tag field. 

here in init()
if 'file_date_range' in self.kwargs:
    # set list files routine to desired date range
    # attach to the instrument object
    self._list_rtn = functools.partial(list_files, 
                                         file_date_range=self.kwargs['file_date_range'])
    self.files.refresh()

and modify the list_files to be
def list_files(tag=None, sat_id=None, data_path=None, format_str=None,
                     file_date_range=None):
    etc.

Comment (rstoneback) [reply pysat/instruments/pysat_testing.py]: It stripped of all the spaces in the code.

File: pysat/instruments/pysat_testing.py (hunk 1 above)
Comment (aburrell) [reply pysat/instruments/pysat_testing.py]: fixed

File: pysat/instruments/pysat_testing.py (hunk 2 above)
Comment (aburrell) [reply pysat/instruments/pysat_testing.py]: I like this solution better than what I did.  I changes things, using this as an example.

**Comment by aburrell** [comment]
The OCB tests in ocbpy run fine with this new date range implementation.

**Comment by aburrell** [comment]
Looks like only the expected tests are failing.

File: pysat/instruments/pysat_testing.py [hunk 3]
```diff
@@ -73,64 +70,109 @@
...
+meta['int16_dummy'] = {'units': '', 'long_name': 'int16_dummy'}
+meta['int32_dummy'] = {'units': '', 'long_name': 'int32_dummy'}
+meta['int64_dummy'] = {'units': '', 'long_name': 'int64_dummy'}
 
 
         
 def init(self):
-    self.new_thing=True        
-                
+    """ Initialization function
```
Comment (rstoneback) [inline pysat/instruments/pysat_testing.py]: Thank you for updating the docstrings.

File: pysat/instruments/pysat_testing.py [hunk 4]
```diff
@@ -73,64 +70,109 @@
...
+        Instrument satellite ID (accepts '')
+    sim_multi_file_right : (boolean)
+        Adjusts date range to be 12 hours in the future or twelve hours beyond
+        root_date (default=False)
+    sim_multi_file_left : (boolean)
+        Adjusts date range to be 12 hours in the past or twelve hours before
+        root_date (default=False)
+    root_date : (NoneType)
+        Optional central date, uses test_dates if not specified.  (default=None)
+    file_date_range : (pds.date_range or NoneType)
```
Comment (rstoneback) [inline pysat/instruments/pysat_testing.py]: Did the code object without the file_date_range keyword in load?

Comment (rstoneback) [reply pysat/instruments/pysat_testing.py]: by code I mean pysat

**Comment by rstoneback** [comment]
Tests confirmed. Only issues are downloads.

File: pysat/instruments/pysat_testing.py (hunk 3 above)
Comment (aburrell) [reply pysat/instruments/pysat_testing.py]: One file at a time :-P

File: pysat/instruments/pysat_testing.py (hunk 4 above)
Comment (aburrell) [reply pysat/instruments/pysat_testing.py]: Yes it did.

**Comment by aburrell** [comment]
Great, I'm merging!  Once we get a minor version release out, I can test and merge my ocbpy commits.

File: pysat/instruments/pysat_testing.py (hunk 4 above)
Comment (rstoneback) [reply pysat/instruments/pysat_testing.py]: Good. I was worried that having keywords used by both init and load could lead to potential collisions or confusion.

**Comment by rstoneback** [comment]
We are getting through the 2.0 milestones. Hopefully not much longer.

**Pull request merged**